public/js/util/validateField.js

```diff
@@ -0,0 +1,25 @@
+const validateField = (fieldName, fieldValue, isRequired: false, customValidation) => {
+  const errors = [];
+  // isRequired check
+  if(isRequired && !fieldValue) {
+    errors.push({
+      error: 'required',
+      message: 'This field is required'
+    });
+  }
+
+  // Custom validators
+  if(customValidation) {
+    customValidation.forEach((validator) => {
+      const result = validator(fieldValue);
+
+      if (result !== true) {
+        errors.push(result);
```

Comment:
Good idea, it'd be good to catch it here rather that it being unrecognised on the field.